### PR TITLE
Store data sets in memory and list them

### PR DESCRIPTION
**Pull request by michaelmattig**
https://trello.com/c/YN9Tmih6

### CODE REVIEW

**Comment by michaelmattig** [comment]
[In-Memory Dataset DB](https://trello.com/c/YN9Tmih6/263-in-memory-dataset-db)

File: services/src/datasets/storage.rs [hunk 1]
```diff
@@ -18,7 +19,21 @@ pub struct DataSet {
...
+    pub source_operator: String,
+    pub spatial_reference: SpatialReferenceOption,
+}
+
+impl DataSet {
+    pub fn listing(&self) -> DataSetListing {
+        DataSetListing {
+            id: self.id.clone(),
+            name: self.name.clone(),
+            description: self.description.clone(),
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/storage.rs:31]: is it actually necessary to clone all this stuff or could we just give out &str? We serialize it anyway.

File: services/src/datasets/storage.rs [hunk 2]
```diff
@@ -133,7 +152,10 @@ pub enum VectorLoadingInfo {
 
 /// Handling of data sets provided by geo engine internally, staged and by external providers
 #[async_trait]
-pub trait DataSetDB: DataSetProvider + Send + Sync {
+pub trait DataSetDB: DataSetProvider + DataSetProviderDB + Send + Sync {}
+
+#[async_trait]
+pub trait DataSetProviderDB {
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/storage.rs:158]: Maybe you can add some info on public traits and types

File: services/src/datasets/storage.rs [hunk 3]
```diff
@@ -159,6 +179,20 @@ pub trait DataSetDB: DataSetProvider + Send + Sync {
     ) -> Result<&dyn DataSetProvider>;
 }
 
+pub trait DataSetStorer: Send + Sync {
+    type D: Send + Sync;
+}
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/storage.rs:184]: if `D` is the meta data type, maybe you should call it
```suggestion
pub trait DataSetStorer: Send + Sync {
    type MetaData: Send + Sync;
}
```

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/datasets/storage.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/datasets/storage.rs:31]: it's not necessary but I use this function only in the in memory DataSetDB. In real deployments using Postgres backend I would create the DataSetListing directly out of data from the db and then the Listing actually has to own the data

File: services/src/datasets/storage.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/datasets/storage.rs:159]: a0c4a7488f489f31a35aab8b045816f567874a98

File: services/src/datasets/storage.rs (hunk 3 above)
Comment (michaelmattig) [reply services/src/datasets/storage.rs]: a0c4a7488f489f31a35aab8b045816f567874a98

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**